File: app/src/main/java/com/uniquext/android/lightpermissiondemo/MainActivity.java
package com.uniquext.android.lightpermissiondemo;

import android.Manifest;
import android.os.Bundle;
import android.util.Log;

import androidx.appcompat.app.AppCompatActivity;

import com.uniquext.android.lightpermission.LightPermission;
import com.uniquext.android.lightpermission.request.PermissionCallback;
import com.uniquext.android.lightpermission.settings.AppSettingsDialog;

/**
 * 　 　　   へ　　　 　／|
 * 　　    /＼7　　　 ∠＿/
 * 　     /　│　　 ／　／
 * 　    │　Z ＿,＜　／　　   /`ヽ
 * 　    │　　　 　　ヽ　    /　　〉
 * 　     Y　　　　　   `　  /　　/
 * 　    ｲ●　､　●　　⊂⊃〈　　/
 * 　    ()　 へ　　　　|　＼〈
 * 　　    >ｰ ､_　 ィ　 │ ／／      去吧！
 * 　     / へ　　 /　ﾉ＜| ＼＼        比卡丘~
 * 　     ヽ_ﾉ　　(_／　 │／／           消灭代码BUG
 * 　　    7　　　　　　　|／
 * 　　    ＞―r￣￣`ｰ―＿
 * ━━━━━━感觉萌萌哒━━━━━━
 *
 * @author UniqueXT
 * @version 1.0
 * @date 2018/11/28  17:35
 */
public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        findViewById(R.id.textview).setOnClickListener(v -> test());

        findViewById(R.id.next).setOnClickListener(v -> test2());

    }

    private static final String[] READ_PERMS = {
            Manifest.permission.WRITE_EXTERNAL_STORAGE,
            Manifest.permission.READ_EXTERNAL_STORAGE
    };

    private void test() {
        LightPermission.with(this)
                .permissions(READ_PERMS)
                .request(new PermissionCallback() {

                    @Override
                    public void onGranted() {
                        Log.e("####", "有权限");
                    }

                    @Override
                    public void onDenied(String[] permissions) {
                        for (String permission : permissions) {
                            Log.e("####", "没有该权限 " + permission);
                        }
                    }

                    @Override
                    public void onProhibited(String[] permissions) {
                        Log.e("####", "拒绝闭关设置不再请求权限，只能去设置页更改");
                        new AppSettingsDialog.Builder(MainActivity.this)
//                                .title("TITLE")
//                                .message("MESSAGE")
                                .show();
                    }

                });
    }

    private void test2() {
        LightPermission.with(this)
                .permissions(READ_PERMS)
                .grant(() ->Log.e("####", "有权限"))
                .deny(permissions -> {
                    for (String permission : permissions) {
                        Log.e("####", "没有该权限 " + permission);
                    }
                })
                .prohibit(permissions -> Log.e("####", "拒绝闭关设置不再请求权限，只能去设置页更改"))
                .request();
    }

}

File: app/src/main/java/com/uniquext/android/lightpermissiondemo/kt/MainActivity.kt
package com.uniquext.android.lightpermissiondemo.kt

import android.Manifest
import android.content.Intent
import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import com.uniquext.android.lightpermission.LightPermission
import com.uniquext.android.lightpermission.PermissionCallback
import com.uniquext.android.lightpermissiondemo.R
import com.uniquext.android.lightpermissiondemo.SecondActivity
import kotlinx.android.synthetic.main.activity_main.*

/**
 * 　 　　   へ　　　 　／|
 * 　　    /＼7　　　 ∠＿/
 * 　     /　│　　 ／　／
 * 　    │　Z ＿,＜　／　　   /`ヽ
 * 　    │　　　 　　ヽ　    /　　〉
 * 　     Y　　　　　   `　  /　　/
 * 　    ｲ●　､　●　　⊂⊃〈　　/
 * 　    ()　 へ　　　　|　＼〈
 * 　　    >ｰ ､_　 ィ　 │ ／／      去吧！
 * 　     / へ　　 /　ﾉ＜| ＼＼        比卡丘~
 * 　     ヽ_ﾉ　　(_／　 │／／           消灭代码BUG
 * 　　    7　　　　　　　|／
 * 　　    ＞―r￣￣`ｰ―＿
 * ━━━━━━感觉萌萌哒━━━━━━
 *
 * @author UniqueXT
 * @version 1.0
 * @date 2019/4/3  11:40
 */
class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        textview.setOnClickListener { test() }
        next.setOnClickListener { test2() }

    }

    private fun test() {
        LightPermission
                .with(this)
                .permissions(Manifest.permission.CAMERA, Manifest.permission.WRITE_EXTERNAL_STORAGE)
                .result(object : PermissionCallback {

                    override fun onGranted() {
                        Log.e("####", "onGranted")
                    }

                    override fun onDenied(permissions: Array<String>) {
                        for (permission in permissions) {
                            Log.e("####", "onDenied $permission")
                        }
                    }

                    override fun onNoRequest(permissions: Array<String>) {
                        for (permission in permissions) {
                            Log.e("####", "onNoRequest $permission")
                        }
                    }

                })

    }

    private fun test2() {
        startActivity(Intent(this, SecondActivity::class.java))
    }


}